source/docs/software/commandbased/commands-v3/mechanisms.rst

```diff
@@ -0,0 +1,70 @@
+# Mechanisms
+
+.. todo::
+   This article provides a comprehensive guide to Mechanisms, the v3 replacement for Subsystems. It explains the philosophy behind Mechanisms, how to create them, best practices for encapsulation, and how they interact with the command system. The focus should be on teaching teams how to properly structure robot hardware abstractions in v3.
+
+## Creating Mechanisms
+
+.. todo::
+   Describe how to create a Mechanism class. Cover:
+   - Basic Mechanism class structure and constructor
+   - How Mechanisms differ philosophically from v2 Subsystems
+   - When to create a Mechanism vs. other organizational patterns
+   - Naming conventions for Mechanism classes
+   - Where Mechanism classes should live in the project structure
+   - How to register Mechanisms with the command scheduler (if needed)
+
+## Encapsulation Best Practices
+
+.. todo::
+   Explain encapsulation principles for Mechanisms. Cover:
```

Comment:
Love it!

Reply:
Maybe also cover an example of something that doesn't need to be encapsulated in a mechanism, like a vision system.